### ISSUE TITLE
Python 3.6 won't install on Windows 2012 R2

### ISSUE THREAD

**Issue by a4660af2-2cdd-4a60-b68c-8005c56fd360** [python/cpython#73769]
BPO | [29583](https://bugs.python.org/issue29583)
--- | :---
Nosy | @pfmoore, @tjguk, @zware, @eryksun, @zooba

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-02-17.12:58:51.889>
created_at = <Date 2017-02-17.00:03:48.239>
labels = ['OS-windows']
title = "Python 3.6 won't install on Windows 2012 R2"
updated_at = <Date 2017-02-17.12:58:51.888>
user = 'https://bugs.python.org/LaneCampbell'
```

bugs.python.org fields:
```python
activity = <Date 2017-02-17.12:58:51.888>
actor = 'eryksun'
assignee = 'none'
closed = True
closed_date = <Date 2017-02-17.12:58:51.889>
closer = 'eryksun'
components = ['Windows']
creation = <Date 2017-02-17.00:03:48.239>
creator = 'Lane Campbell'
dependencies = []
files = []
hgrepos = []
issue_num = 29583
keywords = []
message_count = 4.0
messages = ['287973', '287974', '287980', '288003']
nosy_count = 6.0
nosy_names = ['paul.moore', 'tim.golden', 'zach.ware', 'eryksun', 'steve.dower', 'Lane Campbell']
pr_nums = []
priority = 'normal'
resolution = 'third party'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue29583'
versions = ['Python 3.6']
```

</p></details>

**Comment by a4660af2-2cdd-4a60-b68c-8005c56fd360** [comment]
When trying to install Python 3.6 using the "python-3.6.0-amd64.exe" installer on Windows 2012 R2 I encounter a failure.  

I'm flagging the options in the installer for:
Customize Installation
Add Python 3.6 to PATH
Leaving all Optional Features selected 
Under Advanced Options I'm checking "Install for all users" and letting Python's installer select "Precompile standard library".  
This is what the installer outputs when it fails: http://pastebin.com/XUzVJ1Y5

I've tried to download KB2999226 & KB3118401 manually but they say they aren't for my version of Windows.  I believe that the Python installer is detecting my version of Windows is 8.1 and isn't installing because of the Windows dependencies also fail since they are for version 8.1 (or at least appear to be).

**Comment by eryksun** [comment]
To install KB3118401 [3], you may first need to install KB2919442 [1] and KB2919355 [2]. The installation order would be as follows:

KB2919442:
    Windows8.1-KB2919442-x64.msu

KB2919355:
    clearcompressionflag.exe
    Windows8.1-KB2919355-x64.msu
    Windows8.1-KB2932046-x64.msu
    Windows8.1-KB2959977-x64.msu
    Windows8.1-KB2937592-x64.msu
    Windows8.1-KB2938439-x64.msu
    Windows8.1-KB2934018-x64.msu

KB3118401:
    Windows8.1-KB3118401-x64.msu

[1]: https://www.microsoft.com/en-us/download/details.aspx?id=42153
[2]: https://www.microsoft.com/en-us/download/details.aspx?id=42334
[3]: https://www.microsoft.com/en-us/download/details.aspx?id=50410

**Comment by a4660af2-2cdd-4a60-b68c-8005c56fd360** [comment]
Thanks Eryk, in case this helps anyone else the system is a new install of Windows 2012 R2 as of this week that I updated with all applicable Windows updates.

I tried your approach and succeeded after got the following for two of your packages:

KB2919442 - http://clicky.strapr.com/2W202P2k363m

KB3118401 - http://clicky.strapr.com/3N301p281b0d

The updates in KB2919355 did install and that seemed to fix the error.  Thank you so much!
